server/broadcast.go

```diff
@@ -192,11 +192,16 @@ func selectOrchestrator(n *core.LivepeerNode, params *streamParameters, cpl core
 
 	for _, tinfo := range tinfos {
 		var sessionID string
+		var balance *core.Balance
 
 		if n.Sender != nil {
 			sessionID = n.Sender.StartSession(*pmTicketParams(tinfo.TicketParams))
 		}
 
+		if n.Balances != nil {
```

Comment:
I don't think we're currently initiating balances on the node if it's started up as a broadcaster, unless I missed that change in this PR. Given that fact, `n.Balances` will always be `nil`. We'll have to place the initalization of the balances outside of the check for `orchestrator` flag. 

https://github.com/livepeer/go-livepeer/blob/fdd74879cbe11d792dcadcfbec217ae1bcf2e680/cmd/livepeer/livepeer.go#L401

Reply:
Fixed in 19bec65

Reply:
One more concern that just came up in my mind.

`cleanupInterval` is orchestrator specific and used to be the SenderMonitor and Balance cleanup interval. 

I'm wondering if there's any danger that if we re-use the same value we are cleaning up balances for `BroadcastSessions` when they should in fact be maintained. Sure removing a session happens when the transcodeLoop times out, but I feel like we should only clean after receiving the httpError since we need to account for latency sending back the response. 

I'm also wondering if the `Cleanup` loop is really needed on the `Broadcaster` side because we could also be removing a `Balance` for a session from the map when the session itself is removed or the stream ends.

Reply:
Updated the PR so that B will not use the cleanup routine - see the following commits:

- 6b0968c: Move cleanup loop start/stop so that the routine is only run for orchestrators
- 5ef2995: Add a `Clear()` method to `core.Balance`
- b30af1a: Clear a session's balance upon removal